substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -427,10 +427,8 @@ impl<T: Config> Pallet<T> {
 		}
 
 		// disable all offending validators that have been disabled for the whole era
-		for (index, disabled) in <OffendingValidators<T>>::get() {
```

Comment:
I feel that his PR is removing generality from the current code without simplifying the code (which usually is the main reason to refactor generalizations away). If we want to all the validators to be disabled right, what if we ensure that *all* bools in `OffendingValidators` are true?

In the future, we may want to have a more fine grained disablement strategy. If we remove this, we may have to add it again.

I guess my point is: the way I see it, we can ensure that all validators are disabled without touching this code and storage item.